### ISSUE TITLE
Newly created folder does not show contents in Carbon

### ISSUE THREAD

**Issue by KaiHoewelmeyer-TomTom** [kaihowl/dotfiles#758]

**Comment by kaihowl** [comment]
Repro:
```
edit .
call input("hit enter")
!mkdir -p hello/ && touch hello/a && touch hello/b
edit hello/
call input("hit enter")
edit .
call input("hit enter")
!rm -rf hello/
call input("hit enter")
!mkdir -p hello/ && touch hello/a && touch hello/b
call input("hit enter")
```

**Comment by kaihowl** [comment]
Could not reproduce the accidental deletion of the 'hello' folder.

But `lua= require('carbon.entry').find('/Users/kaihowl/.dotfiles/hello'):children()` returns an empty array while the folder contains files `a` and `b`. The entry in the cache seems stale.

Next: Is the watcher correctly treating the item?

**Comment by kaihowl** [comment]
The watcher is still in place for the folder `hello`. This is either a problem with libuv and watching a deleted / recreated folder or a problem with carbon's infra.

**Comment by kaihowl** [comment]
The problem seems to lie with a resync triggered for a view with a non-existent root.

Does this maybe create an entry with a non-directory type? This will later then prevent recursing the children.

**Comment by kaihowl** [comment]
Found and seemed to have fixed the error: 
A failure to scan a non-existent folder is treated the same way as encountering an empty folder [here](https://github.com/SidOfc/carbon.nvim/blob/9670efb032c4a43c5aa32a0a7ae56afdbf3c275c/lua/carbon/entry.lua#L143). The folder is then treated as empty and stored in the entries.items list. This way it remains empty.

A fix is to add an `else` case [here](https://github.com/SidOfc/carbon.nvim/blob/9670efb032c4a43c5aa32a0a7ae56afdbf3c275c/lua/carbon/entry.lua#L145) and treat non `userdata` cases / `nil` cases for the `handle` by returning a `nil` value instead of an empty table.

**Comment by kaihowl** [comment]
https://github.com/SidOfc/carbon.nvim/pull/137

**Comment by kaihowl** [comment]
Fixed with carbon in version 0.20.1. Tested and confirmed.
